File: add_item.py
#!/Users/w/opt/anaconda3/bin/python3

print("Content-type:text/html")
print()

import cgi, os
import re
from xml.dom.minidom import TypeInfo
import pymysql
import logging
import cgitb; cgitb.enable()
from PIL import Image
import base64,io



def connectDb(dbName):
    try:
        mysqldb = pymysql.connect(
                host="127.0.0.1",
                user="root",
                port=3306,
                database=dbName)
        return mysqldb
    except Exception as e:
        logging.error('Fail to connection mysql {}'.format(str(e)))
    return None

def popWindow(msg,is_succ):
    addr='nav.php'
    print("<script>")
    print('window.location.href="%s";'%addr)
    print('window.alert("%s")'%msg)
    print("</script>")

inFileData = None
form = cgi.FieldStorage() 
UPLOAD_DIR='Picture/'

SID = form.getvalue('SID')
name = form.getvalue('mealname')
pri = form.getvalue('price')
quan = form.getvalue('quantity')
img = form.getvalue('file')
imgType = form.getvalue('imgType')


# print("file: ",len(img),'<br>')



valid=0

try:
    if name==None:
        assert False, 'Item name is empty.'
    if pri==None:
        assert False, 'Price is empty.'
    if quan==None:
        assert False, 'Quantity is empty.' 
    
except AssertionError as msg:
    popWindow(msg,0)

# print("<h1>none empty</h1>")

try:
    parser=re.compile("[ ]*([1-9][0-9]*)[ ]*") 
    match=parser.fullmatch(pri)
    if match:
        pri=match.group(0)
        if int(pri)<0:
            assert False, 'Price is invalid.'
    else:
        assert False, 'Price is invalid.'
    
    parser=re.compile("[ ]*([1-9][0-9]*)[ ]*") 
    match=parser.fullmatch(quan)
    if match:
        quan=match.group(0)
        if int(quan)<0:
            assert False, 'Quantity is invalid.'
    else:
        assert False, 'Quantity is invalid.'
    valid=1
except AssertionError as msg:
    popWindow(msg,0)


# print("<img src='data:",imgType,";base64,",img,"'/>")
# uploaded_file_path = os.path.join(UPLOAD_DIR, os.path.basename(img))


if valid:
    db=connectDb('test') 
    if db is None:
        print('error')
        exit(0)

    # print("<h1>dbconnected</h1>")
    cursor=db.cursor()
    cursor.execute("SELECT name FROM product WHERE name= %s AND SID = %s", (name, SID))
    rlt = cursor.fetchone()
    if rlt!=None:
        msg='The product name has been registered.'
        popWindow(msg,0)
    else:
        cursor = db.cursor()
        sql="""
        INSERT INTO product
        (SID, name, picture, price, quantity, imgType)
        VALUES (%(SID)s, %(name)s, %(img)s, %(pri)s, %(quan)s, "%(imgType)s")
        """
        sql="""
        INSERT INTO product
        (SID, name, picture, price, quantity, imgType)
        VALUES (%s, %s, %s, %s, %s, %s)
        """

        cursor.execute(sql,[SID,name,img,pri,quan,imgType])
        # cursor.execute(sql,{'SID':SID, 'name':name, 'img':img, 'pri':pri, 'quan':quan, 'imgType':imgType})


        db.commit()
        
        popWindow('Add Success.',1)
        
    db.close()

File: add_item.php
<?php
    // echo '檔案名稱: ' . $_FILES['file']['name'] . '<br/>';
    // echo '檔案類型: ' . $_FILES['file']['type'] . '<br/>';
    // echo '檔案大小: ' . ($_FILES['file']['size'] / 1024) . ' KB<br/>';
    // echo '暫存名稱: ' . $_FILES['file']['tmp_name'] . '<br/>';

?>

<?php
if($_FILES["file"]["size"]!=0){
    $file_name=$_FILES["file"]["tmp_name"];
    // resize image
    $maxDim = 100;
    $file_name = $_FILES['file']['tmp_name'];
    list($width, $height, $type, $attr) = getimagesize( $file_name );
    if ( $width > $maxDim || $height > $maxDim ) {
        $target_filename = $file_name;
        $ratio = $width/$height;
        if( $ratio > 1) {
            $new_width = $maxDim;
            $new_height = $maxDim/$ratio;
        } else {
            $new_width = $maxDim*$ratio;
            $new_height = $maxDim;
        }
        $src = imagecreatefromstring( file_get_contents( $file_name ) );
        $dst = imagecreatetruecolor( $new_width, $new_height );
        imagecopyresampled( $dst, $src, 0, 0, 0, 0, $new_width, $new_height, $width, $height );
        imagedestroy( $src );
        imagepng( $dst, $target_filename ); // adjust format as needed
        imagedestroy( $dst );
    }

    $file = fopen($file_name, "rb");
    //讀入圖片檔資料
    $fileContents = fread($file, filesize($file_name)); 
    //關閉圖片檔
    fclose($file);
    //讀取出來的圖片資料必須使用base64_encode()函數加以編碼：圖片檔案資料編碼
    $fileContents = base64_encode($fileContents);

                                    
    // echo '<img src="data:'.$img_type.';base64,' . $fileContents . '" />';

    $sid = $_REQUEST['SID'];
    $name=$_REQUEST['mealname'];
    $price=$_REQUEST['price'];
    $quant=$_REQUEST['quantity'];
    $img_type=$_FILES["file"]["type"];
    // echo 'name: '.$name;
    
    $formName=sprintf("<input type='hidden' name='mealname' value='%s'>",$name);
    $formImg=sprintf("<input type='hidden' name='file' value='%s'>",$fileContents);
    $formType=sprintf("<input type='hidden' name='imgType' value='%s'>",$img_type);
    
    echo<<<EOT
    <form action="add_item.py" method="post">
        <input type='hidden' name='SID' value='$sid'>
        $formName
        <input type='hidden' name='price' value='$price'>
        <input type='hidden' name='quantity' value='$quant'>
        $formImg
        $formType
    </form>
    <script>
        document.getElementsByTagName('form')[0].submit();
    </script>
    EOT;
}else{
    echo<<<EOT
        <script>
            window.location.href="nav.php"
            window.alert("Image is empty")
        </script>
    EOT;
}
?>